### ISSUE TITLE
In-a-box, FIE integration tests for scrollbound/visibility-based animations

### ISSUE THREAD

**Issue by aghassemi** [ampproject/amphtml#10877]
https://github.com/ampproject/amphtml/pull/10818 has added integration tests for scrollbound and visibilit-based `amp-animations` however we need to find a way to test in in-a-box and FIE cases. 

There reasons this is tricky is:
1- we need to scroll the top page but observe an element inside the 3p iframe
2- we need to measure the calculated `opacity` for an element that's inside an iframe which we do not have direct access to. 
One solution is postMessage but that's not really a true integration tests as the a4a creative is not valid if it runs JS. 
Another crazy idea is using CSS blending by overlaying an element (which lives in the parent) that blends differently based on opacity of the item in the iframe and measure that instead!

**Comment by ampprojectbot** [comment]
This is a high priority issue but it hasn't been updated in awhile. @aghassemi Do you have any updates?

**Comment by aghassemi** [comment]
updating priority, this is not easily doable at the moment

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @aghassemi Do you have any updates?

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @aghassemi Do you have any updates?

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @aghassemi Do you have any updates?

**Comment by choumx** [comment]
This may be covered by @cvializ's new testing infra project.

**Comment by rsimha** [comment]
@cvializ Can this be closed with the new E2E tests?

**Comment by cvializ** [comment]
It should be possible, but I have not tested whether these kinds of tests would work flawlessly as e2e is currently implemented

@sparhami's carousel test code tests scrolling logic, so I am inclined to say likely, but let's verify this is working before we close it.

**Comment by lannka** [comment]
this is already done by #24747
